packages/loader/container-loader/src/containerStorageAdapter.ts

```diff
@@ -120,8 +119,21 @@ export class ContainerStorageAdapter
 			// A callback to ensure we fetch the most updated value of service.policies.summarizeProtocolTree, which could be set
 			// based on the response received from the service after connection is established.
 			() => {
-				this._summarizeProtocolTree =
-					this._summarizeProtocolTree ?? service.policies?.summarizeProtocolTree ?? false;
+				// Determine whether or not container should upload the protocol summary along with the summary.
+				// This is determined based on what value is set for serve policy's summariProtocolTree value or the enableSummarizeProtocolTree
+				// retrievd from the loader options or monitoring context config.
+				const shouldSummarize =
+					service.policies?.summarizeProtocolTree ??
+					this.enableSummarizeProtocolTree ??
+					false;
+
+				if (this._summarizeProtocolTree !== shouldSummarize) {
```

Comment:
@jatgarg The initial value of this_summarizeProtocolTree will alway be undefined now, since we are not setting the value any more in the constructor.
So now the valle is always logged 1time as well as whenever changed.